src/run_program.rs

```diff
@@ -191,7 +188,7 @@ where
     ) -> Result<Cost, EvalErr> {
         let op_atom = self.allocator.buf(op_buf);
         // special case check for quote
-        if op_atom.len() == 1 && op_atom[0] == self.quote_kw {
+        if op_atom.len() == 1 && op_atom == self.quote_kw {
```

Comment:
it looks like you meant to drop the `op_atom.len() == 1` condition here. I don't think there are any tests covering this now